### PR TITLE
optimize cropping

### PR DESCRIPTION
**Pull request by MarinPostma**
Optimize cropping as per @kerollmops suggestion.

### CODE REVIEW

**Review by curquiza — APPROVED** [review]
Thanks!!! 😊

File: meilisearch-http/src/index/search.rs [hunk 1]
```diff
@@ -496,18 +496,19 @@ impl<'a, A: AsRef<[u8]>> Formatter<'a, A> {
 
         tokens
             .map(|(word, token)| {
-                if format_options.highlight && token.is_word() && matcher.matches(token.text()).is_some() {
-                    let mut new_word = String::new();
-                    new_word.push_str(&self.marks.0);
-                    if let Some(match_len) = matcher.matches(token.text()) {
+                if let Some(match_len) = matcher.matches(token.text()) {
+                    if format_options.highlight && token.is_word() {
```
Comment (Kerollmops) [inline meilisearch-http/src/index/search.rs:500]:
```suggestion
                if format_options.highlight && token.is_word() {
                    if let Some(match_len) = matcher.matches(token.text()) {
```

Calling `matches` is more CPU intensive than checking that the token is a word or that we must highlight it.

**Review by Kerollmops — CHANGES_REQUESTED** [review]

**Review by Kerollmops — APPROVED** [review]
Nice change! :)

**Comment by MarinPostma** [comment]
bors merge

**Pull request merged**